operate/client/src/modules/hooks/processDefinitions.ts

```diff
@@ -17,10 +20,24 @@ interface ProcessDefinitionWithIdentifier extends ProcessDefinition {
   label: string;
 }
 
+type ProcessDefinitionSelection =
+  | {kind: 'no-match'}
+  | {kind: 'single-version'; definition: ProcessDefinition}
+  | {
+      kind: 'all-versions';
+      definition: Pick<ProcessDefinition, 'name' | 'processDefinitionId'>;
+    };
+
 function getDefinitionIdentifier(definitionId: string, tenantId?: string) {
   return `${definitionId}--${tenantId ?? DEFAULT_TENANT}`;
 }
 
+function getProcessDefinitionName(
+  definition: Pick<ProcessDefinition, 'name' | 'processDefinitionId'>,
+) {
+  return definition.name ?? definition.processDefinitionId;
+}
```

Comment:
💭 I probably would always compute this property immediately and replace the definition's `name` with it or add an additional property like `label` above. This would ensure that this domain logic is always applied when displaying definition names.

However, we usually do not really do it in other parts of the codebase, so I went with the common pattern and call this function on usage site, i.e. _v2/DiagramPanel/DiagramHeader_.